akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -543,20 +534,25 @@ private[akka] class PersistentShard(
       }
 
     case SaveSnapshotFailure(_, reason) ⇒
-      log.warning("PersistentShard snapshot failure: {}", reason.getMessage)
+      log.warning("PersistentShard snapshot failure: [{}]", reason.getMessage)
 
     case DeleteMessagesSuccess(toSequenceNr) ⇒
-      log.debug("PersistentShard messages to {} deleted successfully", toSequenceNr)
-      deleteSnapshots(SnapshotSelectionCriteria(maxSequenceNr = toSequenceNr - 1))
+      val deleteTo = toSequenceNr - 1
+      val deleteFrom = math.max(0, deleteTo - (keepNrOfBatches * snapshotAfter))
```

Comment:
So the low end of the range is a performance opt to not always delete from 0?

Reply:
that's right, and it's used in Cassandra plugin when https://github.com/akka/akka-persistence-cassandra/pull/394 has been released

Reply:
@patriknw @johanandren what is the consequence if I want to upgade akka-cluster version to the 2.5.17 or newer (let's say the latest 2.5.23) but with akka-persistence-cassandra-0.62? Is it possible or do I have to use minimum of akka-persistence-cassandra-0.90 ?